packages/onboarding/src/step-container-v2/components/StepContainerV2/StepContainerV2.tsx

```diff
@@ -50,15 +50,51 @@ export const StepContainerV2 = ( {
 		[ isMediumViewport, isLargeViewport ]
 	);
 
+	const topBarRef = useRef< HTMLDivElement >( null );
+	const [ topBarHeight, setTopBarHeight ] = useState( 0 );
+
+	const stickyBottomBarRef = useRef< HTMLDivElement >( null );
+	const [ stickyBottomBarHeight, setStickyBottomBarHeight ] = useState( 0 );
+
+	useLayoutEffect( () => {
+		const observer = new ResizeObserver( ( entries ) => {
```

Comment:
No one likes running JS while layouts are changing. Currently every step container has a resize observer whether they want it or not. I wonder if it's possible to only enable this when either the sticky footer is visible or the step has a vertically centred layout?

That would probably need an API change though: a `isVerticallyCentred` prop instead of a child element doing the vertical centering.